pkg/resourcemanager/controller/node/reconciler.go

```diff
@@ -57,7 +67,115 @@ func (r *Reconciler) Reconcile(reconcileCtx context.Context, req reconcile.Reque
 		return reconcile.Result{}, nil
 	}
 
-	// TODO: add business logic
+	// prep for checks: list all DaemonSets and all node-critical pods on the given node
+	daemonSetList := &appsv1.DaemonSetList{}
+	if err := r.Client.List(ctx, daemonSetList); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed listing node-critical DaemonSets on node: %w", err)
+	}
+
+	nodeCriticalPodList := &corev1.PodList{}
```

Comment:
```suggestion
	podList := &corev1.PodList{}
```

I found the other name more confusing than it helped, though, this might be a personal preference.